google-cloud/terraform/main.tf

```diff
@@ -0,0 +1,94 @@
+resource "google_service_account" "xtdb_service_account" {
+  project = var.project_id
+
+  account_id   = var.service_account_name
+  display_name = "XTDB Service Account"
+}
+
+resource "google_service_account_iam_binding" "xtdb_service_account_binding" {
```

Comment:
We do ahead of time create a service account binding to allow workload identity federation on the Kubernetes Service Account (prior to actually deploying the kubernetes namespace or kubernetes service account) - we do similar within Azure, too. 

This is mentioned within the docs for both _but_ I wonder if there's a better way to go about it.

Reply:
We _may_ be able to auth the whole cluster/namespace as a bucket admin (from [here](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#verify)) - though I'm not particularly sure about that, either. Feels like IAM service account -> K8s service account makes sense.

Reply:
openly, I'm not following the implications here - lmk if you'd like to go into it in more detail :duck:

Reply:
- Extract namespace and service account out of helm.
- Remove iambinding from terraform,
- Add instructions for setting up namespace/serviceaccount/iam binding via CLI to the doc & guide

Reply:
Perhaps as a pre-install hook?